### ISSUE TITLE
FA102 is missing `collection.abc` types

### ISSUE THREAD

**Issue by Andrej730** [astral-sh/ruff#20512]
### Summary

E.g. use of `Sequence[int]` is not flagged. 
See example below (command I've used is `ruff check test.py --target-version py38 --select=FA102`)

Ruff playground - https://play.ruff.rs/b295c8a4-4295-4a5a-8a2d-e99699814822
```python
from collections.abc import Sequence

# Not reported by Ruff, but results in runtime error:
# TypeError: 'ABCMeta' object is not subscriptable
def test() -> Sequence[int]:
    return [1, 2, 3]


# FA102 Missing `from __future__ import annotations`, but uses PEP 585 collection
def test2() -> list[int]:
    return [1, 2, 3]
```

### Version

ruff 0.13.1

**Comment by ntBre** [comment]
This makes sense to me, but it will have to be a preview change since it expands the scope of a stable rule.
